src/cart/crt_swim.c

```diff
@@ -55,7 +67,40 @@ CRT_RPC_DEFINE(crt_rpc_swim,  CRT_ISEQ_RPC_SWIM, /* empty */)
 CRT_RPC_DECLARE(crt_rpc_swim_wack, CRT_ISEQ_RPC_SWIM, CRT_OSEQ_RPC_SWIM)
 CRT_RPC_DEFINE(crt_rpc_swim_wack,  CRT_ISEQ_RPC_SWIM, CRT_OSEQ_RPC_SWIM)
 
-uint32_t crt_swim_rpc_timeout;
+uint32_t	 crt_swim_rpc_timeout;
+static bool	 crt_swim_should_fail;
+static uint64_t	 crt_swim_fail_delay;
+static uint64_t	 crt_swim_fail_hlc;
+static swim_id_t crt_swim_fail_id;
+
+static struct d_fault_attr_t *d_fa_swim_drop_rpc;
+
+static void
+crt_swim_fault_init(const char *args)
+{
+	char	*s, *s_saved, *end, *save_ptr = NULL;
+
+	D_STRNDUP(s_saved, args, strlen(args));
+	s = s_saved;
+	if (s == NULL)
+		return;
+
+	while ((s = strtok_r(s, ",", &save_ptr)) != NULL) {
+		while (isspace(*s))
+			s++; /* skip space */
+		if (!strncasecmp(s, "delay=", 6)) {
+			crt_swim_fail_delay = strtoul(s + 6, &end, 0);
+			D_ERROR("*** CRT_SWIM_FAIL_DELAY=%lu\n",
+				crt_swim_fail_delay);
```

Comment:
This is fine for now, but in future we might want to add D_ALWAYS_PRINT or some such instead of hijacking 'error' print for these purposes. We've had situations already when end-users and our devs were confused by errors being emitted by swim that are not errors.

Reply:
Agree. But this print should go into the log file (not just on screen). My assumption about word "print" is just screen output.

Reply:
Correct, it should go to log file as well. In cart tests we use DBG_PRINT() for example that also goes to the log file. We can pick the appropriate naming for this like D_LOG_ALWAYS instead maybe to make it clear.

Reply:
I don't see why a regular D_ERROR is not sufficient here, but we also have CRIT and ALARM I believe.

Reply:
D_ERROR is sufficient, but it is misleading when things show up as 'error' in the log, while we just want to print them always and simply using D_ERROR to force that behavior for non-error messages

Reply:
Even ERR isn't guaranteed to print.   No reason a user couldn't do log_mask: CRIT.  If you actually want it to never be skipped, you will need to add a new type.   We don't have any bits but we don't use ALERT or NOTE anywhere....we could potentially move the bits down, hijack bit 7 and make it so the user can only set EMRG or lower as the mask.   There is still the issue of a user setting D_SUBSYS to disable messages though...